src/Build/Logging/ParallelLogger/ParallelConsoleLogger.cs

```diff
@@ -535,6 +532,58 @@ public override void ProjectStartedHandler(object sender, ProjectStartedEventArg
                     WriteItems(e, e.Items);
                 }
             }
+
+            //node and project context ids for the propertyOutputMap key
+            int node = -1;
+            int projectContextId = -1;
+            if (e.BuildEventContext != null)
+            {
+                node = e.BuildEventContext.NodeId;
+                projectContextId = e.BuildEventContext.ProjectContextId;
+            }
+            //creating the value to be added to the propertyOutputMap
+            StringBuilder LogOutputProperties = new StringBuilder();
+            if (e.BuildEventContext != null && e.Items != null)
+            {
+                foreach (DictionaryEntry item in e.Items)
+                {
+                    ITaskItem itemVal = (ITaskItem)item.Value;
+                    //finding if the LogOutputProperties item has been used
+                    if (string.Equals((string)item.Key, "LogOutputProperties", StringComparison.OrdinalIgnoreCase))
+                    {
+                        //looking for the property value associated with the property key
+                        //Note: the property key is the item value
+                        string value = null;
+                        bool foundProperty = e.GlobalProperties.TryGetValue(itemVal.ItemSpec, out value);
+     
+                        //looking for the property in local properties if it wasn't found in global properties
+                        if (!foundProperty)
+                        {
+                            foreach (DictionaryEntry prop in e.Properties)
```

Comment:
Should the item itself be key-value pairs of things to log, instead of an indirection to a property name, requiring this second lookup?

Reply:
the original idea was to make it as user friendly as possible. Do you think that would make it tedious for people who are trying to output multiple properties?